interfaces/builtin/dotfiles.go

```diff
@@ -0,0 +1,92 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"fmt"
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/interfaces/apparmor"
+	"github.com/snapcore/snapd/snap"
+)
+
+const dotfilesSummary = `allows access to hidden files in the home directory`
+
+const dotfilesBaseDeclarationSlots = `
+  dotfiles:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const dotfilesConnectedPlugAppArmor = `
+# Description: Can access hidden files in user's $HOME. This is restricted
+# because it gives file access to some of the user's $HOME.
+
+# Note, @{HOME} is the user's $HOME, not the snap's $HOME
+`
+
+type dotfilesInterface struct {
+	commonInterface
+}
+
+func (iface *dotfilesInterface) BeforePreparePlug(plug *snap.PlugInfo) error {
+	hasValidAttr := false
+	var sl []interface{}
+	for _, s := range []string{"files", "dirs"} {
+		if err := plug.Attr(s, &sl); err == nil {
+			hasValidAttr = true
```

Comment:
Please ensure that there are no dot files in here. Ideally call `filepath.Clean` and refuse unclean paths?

Note that we don't want to de-reference symbolic links here as it would allow an attacker to craft a symlink $HOME/.evil that points to /root/ and would allow us to create an apparmor rule that allows write to /root (for example)